java/code/src/com/suse/manager/webui/controllers/DownloadController.java

```diff
@@ -378,11 +376,13 @@ public static Object downloadPackage(Request request, Response response) {
         String channel = request.params(":channel");
         String path = "";
         try {
-            URL url = new URL(request.url());
+            URI uri = new URI(request.url());
             // URL decode the path to support ^ in package versions
-            path = URLDecoder.decode(url.getPath(), StandardCharsets.UTF_8);
+            path = uri.getPath();
+
         }
-        catch (MalformedURLException e) {
+        catch (URISyntaxException e) {
+            log.error(String.format("Unable to parse: %s", request.url()));
             halt(HttpStatus.SC_INTERNAL_SERVER_ERROR,
```

Comment:
I know it's not related to your change, but shouldn't the error code be SC_BAD_REQUEST (400) or SC_UNPROCESSABLE_ENTITY (422)? The error is due to the requested URL being wrong, not a server failure.

Reply:
I would keep it for now. We can think about this in an extra PR